loopy/transform/callable.py

```diff
@@ -154,6 +154,81 @@ def __call__(self, target, identifier):
         return None
 
 
+def subarrayrefs_are_equiv(sar1, sar2, knl):
+    """
+    Compares if two instance of :class:`loopy.symbolic.SubArrayRef`s point
+    to the same array region.
+    """
+    from loopy.kernel.function_interface import get_arg_descriptor_for_expression
+
+    return get_arg_descriptor_for_expression(knl, sar1) == (
+            get_arg_descriptor_for_expression(knl, sar2)) and (
+                    sar1.get_begin_subscript(knl) ==
+                    sar2.get_begin_subscript(knl))
```

Comment:
Bounds finding is very expensive.

I suggest that we use a shallower notion of "equivalence", namely
- check that the expression trees match,
- check that the inames have equivalent bounds.

*DISCUSS*

Reply:
Resolution: Demand `==` on the two expression trees.

Reply:
Done in 6f177eb9.